### DISCUSSION TITLE
iPXE cobbler 3.3.7 - get system-name - ipxe automation

### DISCUSSION THREAD

**Discussion by midas-asim** [cobbler/cobbler#3865] [Q&A]
I have a fresh cobbler 3.3.7 installed on Rocky9.5 - works fine.

I am using ipxe (with windows DHCP and DNS). I get ipxe menu with windows11 & Rocky9 profile options and I can deploy them fine. (secure boot is disabled but using UEFI disks)

As I am installing using ipxe menu, the linux machine will be installed with hostname=localhost - which is expected behaviour.

My wishlist below as per priority:

1. Can I get a script for Linux & windows, that can connect to cobbler api, search mac and get correct host name. This way I can change the hostname before I run domain-join commands.

2. As an improvement to above - if cobbler kickstart can check if the system mac-address exists in cobbler systems - then it should run the custom autoinstall / kickstart template for that system.

3. Even better would be the iPXE menu detects if the system exists and auto starts the correct installation option (like is standard pxe boot for linux machines) - you enable netboot and restart the machine - the machine is re-installed automatically without any input if network boot is enabled.

Can anyone help me with the above.

**Comment by badacook** [comment]
&#8195;  As mentioned in your wish list, implement host to name, default anserfile, default menu association and automatic adjustment through rules to achieve more intelligent automatic deployment.It may be necessary to add such rules as data layer abstraction. At present, we only specify the client category in the DHCP configuration, and achieve the customization of bootfiles, which may only be attempted from the product architecture design of cobbler.
&#8195;  In addition, since you have successfully completed the deployment of Windows11 in cobbler 3.3.7, I would like to ask **the specific version number of Windows11** for convenience Cobbler server **command ` grep - A 30 "Available Images"/var/log/cobbler/cobbler. Log ` output**，many thanks.

**Comment by midas-asim** [comment]
Hi,

Windows 11 Pro - version: 23H2 (OS Build 22631.2428) is what I am using in
the office Test Server.

I am not using Secure Boot.

Home lab, I am using more recent version and on Vmware esxi-8, so had to
add the following to answer.template

           <RunSynchronous>
                               <RunSynchronousCommand wcm:action="add">
                                       <Order>1</Order>
                                       <Path>reg.exe add
"HKLM\SYSTEM\Setup\LabConfig" /v BypassTPMCheck /t REG_DWORD /d 1
/f</Path>
                               </RunSynchronousCommand>
                               <RunSynchronousCommand wcm:action="add">
                                       <Order>2</Order>
                                       <Path>reg.exe add
"HKLM\SYSTEM\Setup\LabConfig" /v BypassSecureBootCheck /t REG_DWORD /d
1 /f</Path>
                               </RunSynchronousCommand>
                               <RunSynchronousCommand wcm:action="add">
                                       <Order>3</Order>
                                       <Path>reg.exe add
"HKLM\SYSTEM\Setup\LabConfig" /v BypassRAMCheck /t REG_DWORD /d 1
/f</Path>
                               </RunSynchronousCommand>

           </RunSynchronous>


For Win.ks template to work - I had to change the FirstLogonCommands as
follow in answer.template (this should be standard for all cobbler installs
- but is missing).

           <FirstLogonCommands>^M
               <SynchronousCommand wcm:action="add">^M
                   <RequiresUserInput>false</RequiresUserInput>^M
                   <Order>1</Order>^M
                   <CommandLine>cmd /C wmic useraccount where
"name='LocalUser'" set PasswordExpires=FALSE</CommandLine>^M
               </SynchronousCommand>^M
               <SynchronousCommand wcm:action="add">^M
                   <RequiresUserInput>false</RequiresUserInput>^M
                   <Order>2</Order>^M
                   <CommandLine>powershell -command "Get-NetAdapter |
foreach { Disable-NetAdapterBinding -InterfaceAlias \$_.Name
-ComponentID ms_tcpip6 }"</CommandLine>^M
               </SynchronousCommand>^M
               <SynchronousCommand wcm:action="add">
                   <RequiresUserInput>false</RequiresUserInput>
                   <Order>3</Order>
                   <CommandLine>powershell -command "& {\$WebClient =
New-Object System.Net.WebClient;
\$WebClient.DownloadFile('http://@@http_server@@/cblr/svc/op/autoinstall/profile/@@profile_name@@','c:\install.cmd')}
<http://%40%40http_server%40@/cblr/svc/op/autoinstall/profile/@@profile_name@@','c:/install.cmd')%7D>"</CommandLine>
               </SynchronousCommand>
               <SynchronousCommand wcm:action="add">
                   <RequiresUserInput>false</RequiresUserInput>
                   <Order>4</Order>
                   <CommandLine>c:\install.cmd</CommandLine>
               </SynchronousCommand>
           </FirstLogonCommands>^M


*Further Progress:*

On my Win.ks I have powershell scripts to change hostname and join domain
etc - which work but hostname change is intermittent, because I use cobbler
xmlapi to get system-list, but the api token command runs only once, then
starts timing out. Not sure if its because I can generate token only once
for certain amount of time.

Any further help is appreciated.

The above FIRSTLOGON SECTION should be standard for cobbler installs - If
iPXE is the only way to install windows11 then we need the hostname part
automated !

The output you requested is below:

/var/log/cobbler/cobbler.log:[Daemon] 2024-12-15T00:00:00 - INFO | running:
['/usr/sbin/selinuxenabled']
/var/log/cobbler/cobbler.log:[Daemon] 2024-12-15T00:00:00 - INFO | received
on stdout:
/var/log/cobbler/cobbler.log:[Daemon] 2024-12-15T00:00:00 - INFO |
Automigration NOT executed
/var/log/cobbler/cobbler.log:[Daemon] 2024-12-15T00:00:00 - INFO | 11
breeds and 146 OS versions read from the signature file
/var/log/cobbler/cobbler.log:[Daemon] 2024-12-15T00:00:00 - INFO | Cobbler
startup completed

I think the older logs have been archived automatically - so this is
probably not what u need.

Let me know if you need more info

Regards
Asim


On Wed, 18 Dec 2024 at 09:34, badacook ***@***.***> wrote:

> As mentioned in your wish list, implement host to name, default anserfile,
> default menu association and automatic adjustment through rules to achieve
> more intelligent automatic deployment.It may be necessary to add such rules
> as data layer abstraction. At present, we only specify the client category
> in the DHCP configuration, and achieve the customization of bootfiles,
> which may only be attempted from the product architecture design of cobbler.
> In addition, since you have successfully completed the deployment of
> Windows11 in cobbler 3.3.7, I would like to ask *the specific version
> number of Windows11* for convenience Cobbler server *command grep - A 30
> "Available Images"/var/log/cobbler/cobbler. Log output*，many thanks.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cobbler/cobbler/discussions/3865#discussioncomment-11603146>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BDL4ODMWRQBUPPWEQGK5H5L2GE6UHAVCNFSM6AAAAABTSNZ272VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRQGMYTINQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
The content of this email is strictly confidential and only intended for 
the specified recipient. It is forbidden to share the email or its contents 
with any third party without the sender’s written consent. If this email 
reached you by mistake, please let us know so that we can ensure that this 
doesn’t happen in the future and delete the message.

**Comment by badacook** [comment]
Hi midas-asim

&#8195;  First of all thank for your reply,in fact i just want to see the WIMBoot compatible information in your cobbler.Log about windows image importing ,And have anthoer try. Thanks all the same.
